### PR TITLE
CI Bigblade Vanilla Core PC Coverage

### PR DESCRIPTION
**Pull request by sripathi-muralitharan**
Another portion of what was originally #460  consisting of a nonsynth module for PC coverage and the makefile changes for it.

### CODE REVIEW

**Comment by sripathi-muralitharan** [comment]
This PR is now ready for review.

**Comment by sripathi-muralitharan** [comment]
@dpetrisko this is one of the PRs that requires @tommydcjung's attention.

**Comment by tommydcjung** [comment]
hi @sripathi-muralitharan, could you rebase your changes on the master branch and add a readme on how to run your regression?

**Comment by sripathi-muralitharan** [comment]
Can one of the maintainers change the target branch to master? Would make it easier to review and view the diffs.

**Comment by drichmond** [comment]
Done!

**Comment by sripathi-muralitharan** [comment]
@tommydcjung I rebased the branch and added documentation for the coverage analysis in `software/spmd/README.md`. I also noticed that there are some coverage analysis targets and code that could be salvaged here -  https://github.com/bespoke-silicon-group/bsg_manycore/blob/c616db578bd3a9d3be9f5d1962eeabc703b05121/software/spmd/Makefile#L40-L104. Let me know if we can fold this into this PR as well and what needs to be done.

File: testbenches/common/v/bsg_nonsynth_manycore_testbench.v
Comment (tommydcjung) [inline testbenches/common/v/bsg_nonsynth_manycore_testbench.v:785]: Instead of using enable signal, we really want to use parameter to control its instantiation (like the other profilers), because this will get instantiated in every single tile, and it can significantly slow down the simulation, even when the enable is disabled.

**Comment by tommydcjung** [comment]
> @tommydcjung I rebased the branch and added documentation for the coverage analysis in `software/spmd/README.md`. I also noticed that there are some coverage analysis targets and code that could be salvaged here -
> 
> https://github.com/bespoke-silicon-group/bsg_manycore/blob/c616db578bd3a9d3be9f5d1962eeabc703b05121/software/spmd/Makefile#L40-L104
> Let me know if we can fold this into this PR as well and what needs to be done.

Not required for this PR, but if you can salvage something, go for it.

File: testbenches/common/v/bsg_nonsynth_manycore_testbench.v
Comment (sripathi-muralitharan) [reply testbenches/common/v/bsg_nonsynth_manycore_testbench.v]: Addressed in the latest commit.

**Comment by sripathi-muralitharan** [comment]
@tommydcjung is there any more work required on this before it can be merged?

**Review by tommydcjung — APPROVED** [review]

File: software/spmd/interrupt_tests/Makefile
Comment (tommydcjung) [inline software/spmd/interrupt_tests/Makefile:26]: LOG_FILES needs to change also. *_run.log -> *.run.log

Comment (sripathi-muralitharan) [reply software/spmd/interrupt_tests/Makefile:26]: Modified LOG_FILES and SCRAPE-TAIL to reflect the requested change.

**Comment by tommydcjung** [comment]
I think there is merge conflict to master branch.

**Comment by sripathi-muralitharan** [comment]
I can't see what the conflicts are. The GUI doesn't show and I rebased in my local branch and it seems to be fine. It also looks like you merged the PR but this one shows it is still open? (I refreshed the screen a bunch of times)

**Comment by tommydcjung** [comment]
Yeah, I'm not sure what happened with github, but it seems like all your commits are merged.

**Pull request merged**
